src/wallet/walletutil.h

```diff
@@ -47,6 +47,9 @@ enum WalletFlags : uint64_t {
     // Indicates that the descriptor cache has been upgraded to cache last hardened xpubs
     WALLET_FLAG_LAST_HARDENED_XPUB_CACHED = (1ULL << 2),
 
+    // Indicates that the wallet is capable of having a wallet hd key
+    WALLET_FLAG_GLOBAL_HD_KEY = (1ULL << 3),
```

Comment:
Older node has no concept of master key for descriptors wallets, so it can't rotate the key. Can it?

Reply:
When it encrypts the wallet I think it generates a new set of descriptors based on a new master key.

Reply:
Yes, encrypting is the only way to rotate a master key currently.

The issue with changing this to a required flag is that it can no longer be an automatic upgrade. It will need to be an explicit upgrade.